clash-ghc/src-ghc/Clash/GHC/LoadModules.hs

```diff
@@ -330,6 +330,10 @@ loadModules useColor hdl modName dflagsM idirs = do
   startTime <- Clock.getCurrentTime
   GHC.runGhc (Just libDir) $ do
     setupGhc useColor dflagsM idirs
+    -- TODO: We currently load the transitive closure of _all_ bindings found
+    -- TODO: in the top module. This is wasteful if one or more binders don't
+    -- TODO: contribute to any top entities. This effect is worsened when using
+    -- TODO: -main-is, which only synthesizes a single top entity.
```

Comment:
Not true any longer? the bit about the single top entity

Reply:
We still load _all_ potential top entities and their dependency closure, even though in the end we might only pick out one (and _its_ dependencies) to synthesize. The last sentence is a bit misleading: maybe it should read "which only synthesizes a single top entity (and any top entities it depends on)".

Reply:
Changed the last sentence.